src/axom/core/ArrayBase.hpp

```diff
@@ -475,7 +629,7 @@ bool operator==(const ArrayBase<T1, DIM, LArrayType>& lhs,
 
   for(int i = 0; i < static_cast<const LArrayType&>(lhs).size(); i++)
   {
-    if(!(lhs[i] == rhs[i]))
+    if(!(lhs.flatIdx(i) == rhs.flatIdx(i)))
     {
       return false;
     }
```

Comment:
Minor: Do you expect noticeably overhead from using the `flatIdx()` here instead of directly accessing the elements?

I suppose I'm wondering how much of this gets optimized/compiled out and which parts are not optimized.

**Edit:** Clearly, it's not worse than it was before this PR